docs/source/conf.py

```diff
@@ -188,7 +186,7 @@
 autodoc_typehints = "description"
 autodoc_default_options = {
     "members": True,
-    "inherited-members": True,
+    "inherited-members": "int",
```

Comment:
I'm not sure about this attribute, but when this value is `True` with python 3.12, we got the following warning messages:

```shell
optuna/study/_study_direction.py:docstring of optuna.study.StudyDirection.from_bytes:9: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
optuna/study/_study_direction.py:docstring of optuna.study.StudyDirection.to_bytes:8: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
optuna/trial/_state.py:docstring of optuna.trial.TrialState.from_bytes:9: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
optuna/trial/_state.py:docstring of optuna.trial.TrialState.to_bytes:8: WARNING: Inline interpreted text or phrase reference start-string without end-string. [docutils]
```

As explained by https://stackoverflow.com/questions/77914856/sphinx-raises-warnings-for-class-derived-from-intenum-can-i-do-something-about, which was shared by @himkt privately, the workaround is that wait for python 3.13 or use `replace `True` with int` as in this line. 

I appreciate any comments on this.

## Master

![Screenshot 2024-11-02 at 15 15 40](https://github.com/user-attachments/assets/145dc6a2-e734-4794-9071-5595343f6a19)

## This PR

![Screenshot 2024-11-02 at 15 15 22](https://github.com/user-attachments/assets/fcc1c1b4-bc48-4497-8c2c-c2e500b69f21)

Reply:
It seems that the page of `optuna.trial.TrialState` as you mentioned with the current PR is completely same as that with the master branch.

https://optuna--5742.org.readthedocs.build/en/5742/reference/generated/optuna.trial.TrialState.html#optuna.trial.TrialState

<img width="942" alt="Screenshot 2024-11-06 at 14 47 27" src="https://github.com/user-attachments/assets/028e47e9-e4cb-4ea5-b1ac-4e3b42389dbe">

Do I misunderstand something?

Reply:
Interesting. It sounds like my local Sphinx build issue. Never mind.

Reply:
I think that ReadTheDocs did not use Python 3.12 because `.readthedocs.yml` was not updated.
https://github.com/optuna/optuna/blob/9def06d79213526d565eada4612eeeedbc2db060/.readthedocs.yml#L11

Reply:
Thank you for letting us know that. Should we update this too?

Reply:
I've updated python version in `.readthedocs.yml`. Then `optuna.trial.TrialStat` page is different from the master's docs as I written above.

https://optuna--5742.org.readthedocs.build/en/5742/reference/generated/optuna.trial.TrialState.html#optuna.trial.TrialState